### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by jheart-vic**
- Created class `Person` with  the following:
  - Instance vars: `@id`, `@name`, and `@age`.
  - Constructor with `name`,  `age`, and `parent_permission` as parameter. `name` and `parent_permission` are optional and have default values of `"Unknown"` and `true`.
  - Getters for `@id`, `@name`, and `@age`.
  - Setters for `@name` and `@age`.
  - Private method `is_of_age?` that returns `true` if `@age` is greater or equal to 18 and `false` otherwise.
  - Public method `can_use_services?` that returns `true` if person is of age or if they have permission from parents.
- Created class `Student` with the following:
  - Inherits from `Person`.
  - Constructor extends parent's constructor by adding `@classroom` and a parameter for it.
  - Method `play_hooky` that returns `"¯\(ツ)/¯"`.
- Created class `Teacher` with the following:
  - Inherits from `Person`.
  - Constructor extends parent's constructor by adding `@specialization` and a parameter for it.
  - Override `can_use_services?` so it always returns `true`.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    @age = age
+  end
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    of_age?
+  end
```
Comment (FrederickMih) [inline person.rb]: - [ ] Hi @jheart-vic, based on the requirements of this project, this public method(can_use_services?) should return true if a person is of age :+1: or if they have permission from parents(`is_of_age?() || @parent_permission`) :angry:  
> **Quick note**
> > You'll equally need to set `parent_permission` as a parameter in the constructor above and default the value to `true` that's 
```ruby   
   def initialize(age, name =  "Unknown", parent_permission = true)
    
      @parent_permission = parent_permission
   end
```

[ReadMore...](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/class_and_object_for_irb.md#constructor-and-instance-vars)

Thanks :pray:

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  private :of_age
+  public :can_use_services
```
Comment (FrederickMih) [inline person.rb:21]: - [**Optional**] I don't think you explicitly need to set this method as public. In Ruby, a method is by default public if not under the `private` keyword.

File: student.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Student < Person
+  def initialize(classroom, name, age)
+    super(name, age)
```
Comment (FrederickMih) [inline student.rb]: - [ ] I think a student will require to obtain permission from the parents if he/she is not of age before renting a particular book. Hence you'll need to pass `parent_permission` as a parameter in the `super()` method to inherit from the parent class(Person) 
- **Hints**

```ruby
def initialize(classroom, name = "Unknown", age, parent_permission = true)  
    super(age, name, parent_permission)
    @classroom = classroom
end
```

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @jheart-vic , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- Correct GitHub workflow :-1: 
- No Linter error :heavy_check_mark: 
- PR has a title and summary :heavy_check_mark: 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mugishasam123 — APPROVED** [review]
Hi @jheart-vic  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors 👍 
- Good commits 💯 

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**